pkg/commands/chgm/chgm.go

```diff
@@ -0,0 +1,190 @@
+package chgm
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/aws"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+
+	"github.com/aws/aws-sdk-go/service/cloudtrail"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+)
+
+// these type aliases are here to make the types unique and unambiguus when using inside the struct
+type AwsClient = aws.Client
+type OcmClient = ocm.Client
+
+// ChgmProvider should have all of the functions that ChgmService is implementing
+type ChgmProvider struct {
+	// having awsClient and ocmClient this way
+	// allows for all of the method receivers defined on them to be passed into the parent struct,
+	// thus making it more composable than just having each func redefined here
+	//
+	// a different solution is to have the structs have unique names to begin with, which makes the code
+	// aws.AwsClient feel a bit redundant
+	// TODO: decide if the client types should be unique or not
+	AwsClient
+	OcmClient
+}
+
+// This will generate mocks for the interfaces in this file
+//go:generate mockgen --build_flags=--mod=readonly -source $GOFILE -destination ./mock/interfaces.go -package mock
+
+// ChgmService will wrap all the required commands the client needs to run it's operations
+type ChgmService interface {
+	// AWS
+	ListStopInstancesEvents() ([]*cloudtrail.Event, error)
+	ListRunningInstances(infraId string) ([]*ec2.Instance, error)
+	ListStoppedInstances(infraId string) ([]*ec2.Instance, error)
+	// OCM
+	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
+	GetClusterInfo(identifier string) (*v1.Cluster, error)
+}
+
+// Client is an implementation of the Interface, and adds functionality above it
+// to create 'Client' you can use a mock, or fill it with
+type Client struct {
```

Comment:
I am not sure if we need this additional struct since ChgmProvider already implements the interface. Lets have a video review/coding sessions tomorrow to do some testing on this.

Reply:
we missed the 1:1 / coding session on this
hopefully this is the last comment before merge 🤞